server/swap/swap.go

```diff
@@ -874,19 +906,27 @@ func bufferedTicker(ctx context.Context, dur time.Duration) chan struct{} {
 }
 
 func (s *Swapper) tryConfirmSwap(ctx context.Context, status *swapStatus, confTime time.Time) (final bool) {
-	status.mtx.Lock()
-	defer status.mtx.Unlock()
-	if status.swapTime.IsZero() || !status.swapConfirmed.IsZero() {
-		return
+	if known, confirmed := status.contractState(); !known {
+		return // no swap yet to confirm
+	} else if confirmed {
+		return true // already confirmed
 	}
+
+	// Swap known means status.swap is set, and that it will not be replaced
+	// because we are gating processInit with the swapSearching semaphore.
```

Comment:
Note that an easy way to just avoid a data race would have been to just use a getter method like the other swapStatus methods, but as I'm pointing out it's is completely incorrect for the swap contract field to be changed once set so I wanted to fix that issue properly.